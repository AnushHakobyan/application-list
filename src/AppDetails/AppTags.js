import React from 'react';
import PropTypes from 'prop-types';

const AppTags = ({ tags }) => (
  <div className="tags">
    {tags.map(tag => (
      <span key={tag}>
        <span>{tag}</span>
        /
      </span>
    ))}
  </div>
);

AppTags.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.string).isRequired,
}

export default AppTags;
