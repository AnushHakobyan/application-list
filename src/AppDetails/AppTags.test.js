import AppTags from './AppTags';

describe('AppTags Component', () => {
  it ('should render tags and slashes between them', () => {
    const wrapper = shallow(<AppTags tags={['tag 1', 'tag 2', 'tag 3']} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('[data-testid="separator"]')).toHaveLength(2);
  }),
  it ('should not render a slash at the end of or at the beginning of tags', () => {
    const wrapper = shallow(<AppTags tags={['single tag']} />);
    expect(wrapper.find('[data-testid="separator"]')).toHaveLength(0);
  })
});
