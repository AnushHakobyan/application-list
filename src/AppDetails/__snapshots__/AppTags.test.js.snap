// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTags Component should render tags and slashes between them 1`] = `
<div
  className="tags"
>
  <span
    key="tag 1"
  >
    <span>
      tag 1
    </span>
  </span>
  <span
    key="tag 2"
  >
    <span
      data-testid="separator"
    >
      '/'
    </span>
    <span>
      tag 2
    </span>
  </span>
  <span
    key="tag 3"
  >
    <span
      data-testid="separator"
    >
      '/'
    </span>
    <span>
      tag 3
    </span>
  </span>
</div>
`;
